core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -98,7 +98,10 @@ func (ec *ethConfirmer) Disconnect() {
 // - Now we finish head 41
 // - We move straight on to processing head 45
 func (ec *ethConfirmer) OnNewLongestChain(ctx context.Context, head models.Head) {
-	ec.mb.Deliver(head)
+	wasOverCapacity := ec.mb.Deliver(head)
+	if wasOverCapacity {
+		logger.Error("EthConfirmer: head mailbox is over capacity - dropped the oldest unprocessed head")
```

Comment:
Maybe check with @samsondav how important this is. I would think `Warn`.

Reply:
True, i checked that it's only used a limit, so each head is not needed directly